src/coreclr/src/vm/methodtablebuilder.cpp

```diff
@@ -11602,6 +11602,23 @@ BOOL MethodTableBuilder::ChangesImplementationOfVirtualSlot(SLOT_INDEX idx)
         // methods.
         if (!fChangesImplementation && (ParentImpl.GetSlotIndex() != idx))
             fChangesImplementation = TRUE;
+
+        if (!fChangesImplementation)
+        {
+            // Search the previous slots in the parent vtable for the same implementation. If it exists and it was
+            // overriden, the ClassLoader::PropagateCovariantReturnMethodImplSlots will propagate the change to the current
+            // slot (idx), so the implementation of it will change.
+
+            MethodDesc* pParentMD = ParentImpl.GetMethodDesc();
+            for (SLOT_INDEX i = 0; i < idx; i++)
```

Comment:
@davidwrighton one thing I was considering here is to run the lookup loop only when the PreserveBaseOverridesAttribute is found on the pParentMD. But I was not sure what is the performance profile of the `MethodDesc::GetCustomAttribute(WellKnownAttribute::PreserveBaseOverridesAttribute, ...)`.

Reply:
Ah, I haven't realized before that I could call the RequiresCovariantReturnTypeChecking on the pParentMD and that check is cheap.

Reply:
I've added commit with such a change.